### PR TITLE
Add `run_with_call_header` field to `CallTrace`

### PR DESCRIPTION
**Pull request by ddoktorski**
Related https://github.com/software-mansion/cairo-profiler/issues/75

## Introduced changes

- Updated profiler related code to align with the changes from https://github.com/software-mansion/cairo-profiler/pull/92

## Checklist

- [X] Linked relevant issue
- [ ] Updated relevant documentation
- [ ] Added relevant tests
- [X] Performed self-review of the code
- [ ] Added changes to `CHANGELOG.md`

### CODE REVIEW

File: crates/forge/tests/e2e/build_profile.rs [hunk 1]
```diff
@@ -2,6 +2,7 @@ use super::common::runner::{setup_package, test_runner};
 use forge_runner::profiler_api::PROFILE_DIR;
 
 #[test]
+#[ignore]
```
Comment (piotmag769) [inline crates/forge/tests/e2e/build_profile.rs:5]: Make an issue to uncomment it when we release a new profiler

Comment (ddoktorski) [reply crates/forge/tests/e2e/build_profile.rs:5]: https://github.com/software-mansion/cairo-profiler/issues/93

File: crates/cheatnet/src/state.rs [hunk 2]
```diff
@@ -184,8 +185,9 @@ impl CairoSerialize for CallTrace {
 }
 
 impl CallTrace {
-    fn default_successful_call() -> Self {
+    fn default_successful_call(run_with_call_header: bool) -> Self {
         Self {
+            run_with_call_header,
```
Comment (Draggu) [inline crates/cheatnet/src/state.rs:190]:
```suggestion
    fn default_successful_call() -> Self {
        Self {
            run_with_call_header: Default::default(),
```

File: crates/cheatnet/src/state.rs [hunk 3]
```diff
@@ -331,7 +333,7 @@ impl Default for CheatnetState {
         test_code_entry_point.class_hash = Some(class_hash!(TEST_CONTRACT_CLASS_HASH));
         let test_call = Rc::new(RefCell::new(CallTrace {
             entry_point: test_code_entry_point,
-            ..CallTrace::default_successful_call()
+            ..CallTrace::default_successful_call(true)
```
Comment (Draggu) [inline crates/cheatnet/src/state.rs:336]:
```suggestion
            run_with_call_header: true,
            ..CallTrace::default_successful_call()
```

File: crates/cheatnet/src/state.rs [hunk 4]
```diff
@@ -459,7 +461,7 @@ impl TraceData {
     ) {
         let new_call = Rc::new(RefCell::new(CallTrace {
             entry_point,
-            ..CallTrace::default_successful_call()
+            ..CallTrace::default_successful_call(false)
```
Comment (Draggu) [inline crates/cheatnet/src/state.rs:464]:
```suggestion
            ..CallTrace::default_successful_call()
```

**Review by Draggu — APPROVED** [review]
Please check comments before merging

**Pull request merged**